### ISSUE TITLE
When the contract isn't verified I get an error

### ISSUE THREAD

**Issue by sjunaiduk** [0xweb-org/0xweb#2]
05:20:00 ABI is not resolved from ....9f6aE4670183662F96ecd97E: Contract source code not verified. Extract from bytecode...
(node:6360) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
Provided address undefined is invalid, the capitalization checksum test failed, or it's an indirect IBAN address which can't be converted.

**Comment by tenbits** [comment]
Hey @sjunaiduk what is the exact command you run? E.g. `0xweb install 0x... --name Foo --chain eth` will try to fetch the validated sources for the `0x...` address from etherscan. If it fails, it will try to fetch the bytecode from the blockchain and extract the `methods` from it. _But the bytecode extractor, is not ideal, as the method signatures are taken from [functions.json](https://github.com/0xweb-org/dequanto/blob/90aa835f606ae1d2b16d5f72500cb7bf45ed4f17/data/evm/functions.json)_
If the contract is not validated you can generate the classes with ABI in json file: `0xweb install ./contract/mycontract.json --name Foo --chain eth` and later on you must provide the address when initializing the code:

```typescript
import { Foo } from '@0xweb/eth/Foo'
//..
let foo = new Foo('0x...');
```

**Comment by tenbits** [comment]
Seems the issue has been fixed in https://github.com/0xweb-org/dequanto/commit/095ea49dd44da31633a155f00d6208c29713c428

This fix is also published to npm in `0xweb@0.9.62`

@sjunaiduk Could you please check?

**Comment by sjunaiduk** [comment]
Hey, i'm not getting that error anymore. But Im now getting a different one, I basically used the CLI and ran 0xweb install 0x285C5AF84db68098BF8aBDbdF9121791460F13fF --name copyfi --chain bsc   and got this result:

![image](https://user-images.githubusercontent.com/51975746/228581603-ad487fa2-7575-440e-8ec5-197c7e08e46c.png)






I'm trying to interact with it by typing "0xweb contract copyfi (methodNameHere" but i get this error:

![image](https://user-images.githubusercontent.com/51975746/228582605-a0ea3a5b-a0d6-4ff4-9130-0e69a2b63345.png)



I also checked the fetched methods from the byte code;
![image](https://user-images.githubusercontent.com/51975746/228583463-d59cd485-e5ea-4dd6-a5e7-5f9bf2b5f741.png)

It doesn't match with what I see on ethervm.io

![image](https://user-images.githubusercontent.com/51975746/228583519-aaa14d0c-3d4b-4b29-bfb0-a90873bca0e8.png)

**Comment by tenbits** [comment]
Should now be fixed (in `0xweb@0.9.69`)

```bash
$ 0xweb install 0x285C5AF84db68098BF8aBDbdF9121791460F13fF --name copyfi --chain bsc
$ 0xweb c read copyfi decimals
# or you can provide the ABI to read
$ 0xweb c read copyfi "decimals() returns uint256"
```

> The command to read the contracts methods: `0xweb c read contractName contractMethodOrABI`

**Comment by tenbits** [comment]
Closing inactive issue, please reopen if you experience the problem again
